ext/oauth-client/CRM/OAuth/BAO/OAuthContactToken.php

```diff
@@ -2,4 +2,121 @@
 
 class CRM_OAuth_BAO_OAuthContactToken extends CRM_OAuth_DAO_OAuthContactToken {
 
+  /**
+   * Create or update OAuthContactToken based on array-data
+   *
+   * @param array $record
+   * @return CRM_OAuth_DAO_OAuthContactToken
+   */
+  public static function create($record) {
+    self::fillAndValidate($record, CRM_Core_Session::getLoggedInContactID());
+    return static::writeRecord($record);
+  }
+
+  /**
+   * @param $id
+   * @return CRM_OAuth_BAO_OAuthContactToken
+   * @throws CRM_Core_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  public static function del($id) {
+    $record = ['id' => $id];
+    self::fillAndValidate($record, CRM_Core_Session::getLoggedInContactID());
+    return static::deleteRecord($record);
+  }
+
+  /**
+   * @param string $entityName
+   * @param string $action
+   * @param array $record
+   * @param $userId
+   * @return bool
+   * @see CRM_Core_DAO::checkAccess
+   */
+  public static function _checkAccess(string $entityName, string $action, array $record, $userId): bool {
+    try {
+      $record['check_permissions'] = TRUE;
+      self::fillAndValidate($record, $userId);
+      return TRUE;
+    }
+    catch (\Civi\API\Exception\UnauthorizedException) {
```

Comment:
It doesn't seem to like this. It wants a variable like `\Civi\API\Exception\UnauthorizedException $ue`. The error is `ParseError: syntax error, unexpected ')', expecting '|' or variable (T_VARIABLE)`. Not sure if this is php-version specific but I was using 7.4.

Reply:
Good "catch" @demeritcowboy. Fixed, thanks.